src/base.jl

```diff
@@ -1154,3 +1154,69 @@ function _create_system_data_from_kwargs(; kwargs...)
         time_series_in_memory = time_series_in_memory,
     )
 end
+
+"""
+Converts a Line component to a MonitoredLine component and replaces the original in the
+system
+"""
+function convert_component!(
+    linetype::Type{MonitoredLine},
+    line::Line,
+    sys::System;
+    kwargs...,
+)
+    new_line = linetype(
+        line.name,
+        line.available,
+        line.activepower_flow,
+        line.reactivepower_flow,
+        line.arc,
+        line.r,
+        line.x,
+        line.b,
+        (from_to = line.rate, to_from = line.rate),
+        line.rate,
+        line.anglelimits,
+        line.services,
+        line.ext,
+        line._forecasts,
```

Comment:
Time series data is stored with the component UUID, so this won't work unless you use the same UUID for the new line.  You I don't think it would be bad to do that.  Just pass line.internal to the constructor.  Otherwise, you will need to iterate over the line's forecasts, removing and adding each.